### DISCUSSION TITLE
Add support esm modules

### DISCUSSION THREAD

**Discussion by luas10c** [eslint/eslint#17691] [General] [answered]
As most of you may know, nodejs already offers native support for "ESM" I would like to suggest changes to support esm in the configuration file.

**Comment by mdjermanovic** [comment, marked as answer]
Hi, @luas10c.

The new config system supports ESM configuration files:

https://eslint.org/docs/latest/use/configure/configuration-files-new

**Reply by luas10c** [reply]
This flat configuration system is different from the standard configuration.

**Reply by luas10c** [reply]
I didn't understand, how to extend configurations.

**Reply by Rec0iL99** [reply]
The current configuration system (`.eslintrc.*`) will be deprecated in the [`v9.0.0` release](https://eslint.org/blog/2023/10/flat-config-rollout-plans/#flat-config-by-default-in-eslint-v9.0.0) and the flat config system will become the default.

> I didn't understand, how to extend configurations.

Please refer the following docs on how to extend shareable configs in the flat config system.

https://eslint.org/docs/latest/use/configure/configuration-files-new#using-predefined-configurations
https://eslint.org/docs/latest/use/configure/migration-guide#predefined-and-shareable-configs
